scripts/monitoring/cron-send-security-updates-count.py

```diff
@@ -0,0 +1,218 @@
+    config['main']['cachedir'] = ROOT + '/var/cache/yum/$basearch/$releasever'
+    config['main']['reposdir'] = ROOT + '/etc/yum.repos.d'
+    config['main']['installroot'] = ROOT
+    config['main']['logfile'] = '/var/log/hostpkg.yum.log'
+    config.write()
+
+    # edit the repos using fileinput since ConfigParser doesn't support the multi-line format used by yum's baseurl tag
+    for line in fileinput.input(glob.glob(ROOT + '/etc/yum.repos.d/*.repo'), inplace=True):
+        #NOTE: WARNING: do NOT print anything here. STDOUT is being captured
+        #               and written to the repo config file via in-place editing
+        myre = r'^(?P<field>(?:gpgkey|sslcacert|sslclientkey|sslclientcert)\s*=\s*)(?P<path>.+?)(?P<tail>\s*(?:#.*)?)$'
+        match = re.search(myre, line)
+        if match:
+            edited = []
+            # pre-pend ROOT to each path
+            for path in re.split(r',\s*', match.group('path')):
+                match2 = re.search('^(?P<method>(?:[a-z]+://)?)(?P<path>/.*)$', path)
+                if match2 and (not match2.group('method') or match2.group('method') == 'file://'):
+                    edited.append(match2.group('method') + ROOT + match2.group('path'))
+                else:
+                    edited.append(path)
+
+            # edit this line of the file
+            print(match.group('field') + (','.join(edited)) + match.group('tail'), end='')
+        else:
+            # keep the original line
+            print(line, end='')
+
+def parse_args():
+    ''' parse the args from the cli '''
+    parser = argparse.ArgumentParser(description='Count needed software updates and send to monitoring system.')
+    parser.add_argument('-v', '--verbose', default=False, action='store_true',
+                        help='Print more information.')
+
+    return parser.parse_args()
+
+def send(values, options):
+    ''' Send any passed metrics values '''
+    sender = MetricSender()
+
+    if options.verbose:
+        print('Sending values:', values)
+    for key in values:
+        sender.add_metric({key: values[key]})
+    sender.send_metrics()
+
+def main():
+    ''' This script's main function '''
+    logging.basicConfig(format='%(asctime)s - %(relativeCreated)6d - %(levelname)-8s - %(message)s',
+                        filename=LOGFILE,
+                        level=logging.INFO)
+    opts = parse_args()
+    setup()
+
+    logging.info('Cleaning yum metadata')
+    if opts.verbose:
+        print('Cleaning yum metadata')
+    cmd = ['/usr/bin/yum', '--installroot='+ROOT, '-c', ROOT+'/etc/yum.conf', '-q', 'clean', 'metadata']
+    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    stdout, _ = proc.communicate()
+    logging.debug('Metadata cleaned:\n' + stdout)
+    if opts.verbose:
+        print('Metadata cleaned:\n' + stdout)
+
+    secup = SecurityUpdates(opts)
+    # attempt 3 times to get updates.
+    for attempt in range(0, 3):
```

Comment:
should this be variable-ized in case we want to increase attempts? Maybe passed in?